src/coreComponents/physicsSolvers/fluidFlow/IsothermalCompositionalMultiphaseFVMKernels.hpp

```diff
@@ -1125,14 +1118,22 @@ class DiffusionDispersionFaceBasedAssemblyKernel : public FaceBasedAssemblyKerne
         real64 dDiffusionFlux_dC[numFluxSupportPoints][numComp][numComp]{};
         real64 dDens_dC[numComp]{};
 
-        real64 const trans[numFluxSupportPoints] = { stack.transmissibility[connectionIndex][0],
-                                                     stack.transmissibility[connectionIndex][1] };
-
         //***** calculation of flux *****
         // loop over phases, compute and upwind phase flux and sum contributions to each component's flux
         for( integer ip = 0; ip < m_numPhases; ++ip )
         {
-
+          // first, compute the transmissibilities at this face
+          // note that the dispersion tensor is lagged in iteration
+          m_stencilWrapper.computeWeights( iconn,
```

Comment:
By moving this inside the connected elem loops are we looping twice over the connected elems? I think the `SurfaceElementStencil` will loop again because with the star-delta method it needs to know all half transmissibilities to compute the one between two connected cells.